src/v/cloud_storage/tests/topic_recovery_service_test.cc

```diff
@@ -151,13 +156,16 @@ class fixture
     }
 
     using equals = ss::bool_class<struct equals_tag>;
-    void wait_for_n_requests(size_t n, equals e = equals::no) {
-        tests::cooperative_spin_wait_with_timeout(10s, [this, n, e] {
-            if (e) {
-                return get_requests().size() == n;
-            } else {
-                return get_requests().size() >= n;
-            }
+    void wait_for_n_requests(
+      size_t n,
+      equals e = equals::no,
+      std::optional<req_pred_t> predicate = std::nullopt) {
+        tests::cooperative_spin_wait_with_timeout(10s, [this, n, e, predicate] {
```

Comment:
nit: you can use `RPTEST_REQUIRE_EVENTUALLY` so the failed scenario prints the bad line number (instead of just throwing an exception)

Reply:
thanks for the pointer, changed